modules/evm/src/runner/native.rs

```diff
@@ -364,6 +319,29 @@ impl<'vicinity, 'config, T: Trait> Handler<'vicinity, 'config, T> {
 	}
 }
 
+/// Create `try_or_fail` and `try_or_rollback`.
+macro_rules! create_try {
+	( $map_err:expr ) => {
+		macro_rules! try_or_fail {
+			( $e:expr ) => {
```

Comment:
why this needs to be macro? can we make it generic function?

Reply:
Generic function cannot cover the requirement that "if err, return a non err value, else do nothing". Usually `try!` and `?` syntax could handle this scenario but the handler functions doesn't return `Result`.

Reply:
ok. we can refactor the code so it runs in a method return `Result` later